src/common.jl

```diff
@@ -14,4 +14,29 @@ and 1 indicates outside the surface
     iso_vals[7] < iso && (cubeindex |= 0x40)
     iso_vals[8] < iso && (cubeindex |= 0x80)
     cubeindex
+end
+
+"""
+Given a subtype of AbstractMesh, determine the
+type of vertex/point and face to use for internal computations.
+
+Preference is given to the types specified by the Mesh call,
+and will default to the `FieldType` for `SignedDistanceField`,
+and Point{3,Float64}/Face{3,Int} for direct function sampling.
+"""
+function _determine_types(meshtype, fieldtype=Float64)
+        # determine the point and face types
+    # preference is given to the Mesh types
+    # followed by SDF if unspecified
+    if vertextype(meshtype) != Any
```

Comment:
By the way, the compiler often does better with `===` (and `!==`) when comparing types like this, since it's easier to prove that the result is safe to evaluate at compile-time.